File: code/go/2021/aug/patching-array.go
package aug

func minPatches(nums []int, n int) int {
	cnt, patch, tot, i := 0, 0, len(nums), 0
	for patch < n {
		if i < tot && patch+1 >= nums[i] {
			patch += nums[i]
			i++
		} else {
			cnt++
			patch += patch + 1
		}
	}
	return cnt
}

File: archive/python/2021/aug/patching-array.py
from typing import List


class Solutiin:
    def minPatches(self, nums: List[int], n: int) -> int:
        cnt = 0
        patch = 0
        tot = len(nums)
        i = 0
        while patch < n:
            if i < tot and patch + 1 >= nums[i]:
                patch += nums[i]
                i += 1
            else:
                cnt += 1
                # print(patch+1)
                patch += patch + 1  # patch + 1 is the new patch to be added
        return cnt
